crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -32,6 +32,55 @@ def role_constraints
     end
   end
 
+  def shutdown_openstack_at_nodes
+    db_nodes = []
+    NodeObject.all.each do |node|
+      next unless node.state == "crowbar_upgrade"
+
+      # Find the nodes with Database role
+      # (we can't look at node.roles, because that list has been reduced to just crowbar-upgrade)
+      node.crowbar["run_list_map"].each do |role, v|
+        db_nodes.push node.name if role == "database-server"
+      end
+
+      # mark the position in the upgrade process
+      node["crowbar_wall"]["crowbar_openstack_shutdown"] = true
+      node.save
+    end
+
+    # Adapt current proposal, so only non-db nodes have crowbar-upgrade role
+    # (crowbar-upgrade role is currently assigned to all nodes)
+    proposal = ProposalObject.find_proposal("crowbar", "default")
+    proposal["deployment"]["crowbar"]["elements"]["crowbar-upgrade"] -= db_nodes
```

Comment:
Well actually I was wondering what's the reason to not shutdown the non-db service on that cluster anyway? So that crowbar-db-dump would really just be about the database. And remove some of the duplication in the two recipes.